q2_feature_table/_filter.py

```diff
@@ -30,18 +30,23 @@ def f(data_vector, id_, metadata):
 
 
 def _filter(table, min_frequency, max_frequency, min_nonzero, max_nonzero,
-            metadata, where, axis):
+            metadata, where, axis, exclude_ids=False):
     if min_frequency == 0 and max_frequency is None and min_nonzero == 0 and\
-       max_nonzero is None and metadata is None and where is None:
+       max_nonzero is None and metadata is None and where is None and\
+       exclude_ids is False:
         raise ValueError("No filtering was requested.")
     if metadata is None and where is not None:
         raise ValueError("Metadata must be provided if 'where' is "
                          "specified.")
-
+    if metadata is None and exclude_ids is True:
+        raise ValueError("Metadata must be provided if 'exclude_ids' "
+                         "is true.")
```

Comment:
Sorry, I assumed you were familiar with that substitution syntax (that means globally substitute the word `True` for `true`). So, your first option presented, please!